### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by acBerger**
I added some test cases, probably we will need much more, but at the moment I cannot think of new ones.
I am going to work over this and write it out later, but it would be better to first add a few more test cases.
Please tell me if you have some ideas.

### CODE REVIEW

**Comment by RomanLangrehr** [comment]
[PDF](https://github.com/acBerger/Beagle/blob/gh-pages/branches/Testing/Requirements%20Specification.pdf)

**Comment by flyfl** [comment]
I would add a test to verify if the PCM, that was modified by Beagle, is a valid one.

File: doc/Requirements Specification/Parts/Test Cases.lyx [hunk 1]
```diff
@@ -99,12 +110,270 @@ Make sure to stay at the level of acceptance testing.
...
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/T60/"
+
+\end_inset
+
+ Beagle works with different software through
```
Comment (flyfl) [inline doc/Requirements Specification/Parts/Test Cases.lyx]: Isn't this point (T60) quite similar to "Test with Kieker and other measurement software" (T40)?

File: doc/Requirements Specification/Parts/Test Cases.lyx [hunk 2]
```diff
@@ -99,12 +110,270 @@ Make sure to stay at the level of acceptance testing.
...
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/OT20/"
+
+\end_inset
 
+ Detect invalid input
```
Comment (flyfl) [inline doc/Requirements Specification/Parts/Test Cases.lyx]: Who should detect invalid input? The tests or Beagle? (If Beagle -> functional requirement) .
And what kind of inputs? User dialogue between Beagle and user or do you address the source code with "input"?

File: doc/Requirements Specification/Parts/Test Cases.lyx [hunk 3]
```diff
@@ -75,6 +75,17 @@ Test Cases
 \end_layout
 
 \begin_layout Standard
+As Beagle should work with above defined interfaces it has to be tested
+ in complete.
```
Comment (m1cm1c) [inline doc/Requirements Specification/Parts/Test Cases.lyx]: Which interfaces?

"Should" sounds like it's optional. Is this intended?

File: doc/Requirements Specification/Parts/Test Cases.lyx [hunk 4]
```diff
@@ -99,12 +110,270 @@ Make sure to stay at the level of acceptance testing.
...
+
+\begin_layout Labeling
+\labelwidthstring 00.00.0000
+\begin_inset CommandInset label
+LatexCommand label
+name "/T30/"
+
+\end_inset
+
+ Assert that it works for a system with only the software specified in
```
Comment (m1cm1c) [inline doc/Requirements Specification/Parts/Test Cases.lyx]: What "it" is might be obvious for us but not necessarily for the reader of the document.

File: doc/Requirements Specification/Parts/Test Cases.lyx (hunk 2 above)
Comment (m1cm1c) [inline doc/Requirements Specification/Parts/Test Cases.lyx]: Does this conflict with `/C30/`? If it doesn't, it needs more precision.

**Comment by flyfl** [comment]
lgtm

**Comment by jGleitz** [comment]
lgtm

**Pull request merged**